src/app/core/import/import.module.ts

```diff
@@ -1,42 +1,101 @@
+import { EntityAction } from "../entity-details/entity-actions-menu/entity-action.interface";
+import { Entity } from "../entity/model/entity";
+import { Router } from "@angular/router";
+import { EntityTypeLabelPipe } from "../common-components/entity-type-label/entity-type-label.pipe";
+import {
+  AdditionalIndirectLinkAction,
+  AdditonalDirectLinkAction,
+} from "./additional-actions/additional-import-action";
 
 /**
  * UI enabling users to import data from spreadsheets through a guided workflow.
  */
 @NgModule({
   imports: [DiscreteImportConfigComponent],
+  providers: [EntityTypeLabelPipe],
 })
 export class ImportModule {
-  constructor(components: ComponentRegistry) {
-    components.addAll([
-      [
-        "Import",
-        () =>
-          import("./import/import.component").then((c) => c.ImportComponent),
-      ],
-      [
-        "DiscreteImportConfig",
-        () =>
-          import(
-            "../basic-datatypes/discrete/discrete-import-config/discrete-import-config.component"
-          ).then((c) => c.DiscreteImportConfigComponent),
-      ],
-      [
-        "DateImportConfig",
-        () =>
-          import(
-            "../basic-datatypes/date/date-import-config/date-import-config.component"
-          ).then((c) => c.DateImportConfigComponent),
-      ],
-      [
-        "EntityImportConfig",
-        () =>
-          import(
-            "../basic-datatypes/entity/entity-import-config/entity-import-config.component"
-          ).then((c) => c.EntityImportConfigComponent),
-      ],
+  constructor(
+    components: ComponentRegistry,
+    entityActionsMenu: EntityActionsMenuService,
+    private importAdditionalService: ImportAdditionalService,
+    private router: Router,
+    private entityTypeLabelPipe: EntityTypeLabelPipe,
+  ) {
+    components.addAll(importComponents);
+
+    entityActionsMenu.registerActionsFactories([
+      (e) => this.getAdditionalImportActionsForTargetEntity(e),
     ]);
   }
+
+  getAdditionalImportActionsForTargetEntity(entity?: Entity): EntityAction[] {
+    if (!entity) {
+      return [];
+    }
+
+    const actions: EntityAction[] = [];
+    const importActions = this.importAdditionalService
+      .getActionsLinkingTo(entity.getType())
+      .map((a) => ({ ...a, targetId: entity.getId() }));
+
+    for (const importAction of importActions) {
+      let label: string = $localize`Import related ${this.entityTypeLabelPipe.transform(importAction.sourceType, true)} for this ${entity.getConstructor().label}`;
+      if (
+        importActions.filter((a) => a.sourceType === importAction.sourceType)
+          .length > 1
+      ) {
+        // there are multiple imports for the same source type, so we need to distinguish them with additional info:
+        label += ` (${(importAction as AdditonalDirectLinkAction).targetProperty ?? this.entityTypeLabelPipe.transform((importAction as AdditionalIndirectLinkAction).relationshipEntityType)})`;
+      }
```

Comment:
I'm going through the PR changes, and the comments have been really helpful. 
I know that the UX cleanups are still pending, but I wanted to check why, in some cases, `getAdditionalImportActionsForTargetEntity` is showing as Undefined after entity.getConstructor().label.
![image](https://github.com/user-attachments/assets/8f3c94b7-1943-47e0-84cd-fb14ecbfcb16)

Reply:
look into the attached ss; it may be the above exception is related to this(some options are showing empty label)
![image](https://github.com/user-attachments/assets/fb88679d-7bd3-4cd8-ae7b-1f082cdd809b)

Reply:
Thanks! That is indeed weird. I think I need to add some further filters for "disabled" / not fully configured reference fields. Will look into this

Reply:
Managed to fix this now. Thanks.